### ISSUE TITLE
Top[...] makes dict variable after isinstance useless

### ISSUE THREAD

**Issue by ErmoshinMA1-source** [astral-sh/ty#3722]
### Question

I'm confused with the `Top[dict[Unknown, Unknown]]` construct. It essentially makes the dictionary unusable – I cannot call `__getitem__`, I cannot call `.get()`, I cannot in a way narrow the type. There is no way to safely extract a value by key without using `cast`, which defeats the purpose of static checking because it just tells the type checker "trust me".

Am I missing something, or is `cast` really the only way?

```python
        from selenium.webdriver import Chrome

        value = Chrome().find_element().get_property('attributes')
        attributes: dict[str, str] = {}
        if isinstance(value, dict):
            attributes[str(value.get('name', default='default'))] = str(value.get('name', default='default'))

# Method `__getitem__` of type `bound method Top[dict[Unknown, Unknown]].__getitem__(key: Never, /) -> object` cannot be called with key of type `Literal["name"]` on object of type `Top[dict[Unknown, Unknown]]`ty[invalid-argument-type](https://ty.dev/rules#invalid-argument-type)
```

### Version

0.0.46

**Comment by carljm** [comment]
Thanks for the report! This is #2374.

It is accurate that there are very few safe things you can do with an object that is known to be "some kind of dict" but otherwise nothing is known about it. You don't even know what type of keys it has.

That said, there are some safe things you can do. You can iterate its keys/values/items (typed as `object`), and we do allow that. It is also safe to call `.get()` on it, since that method doesn't raise even if an invalid key is given. And that's what you do here.

Unfortunately typeshed's type annotation of `dict.get` is (IMO) overly strict here, and that causes us to error here. I think we should either override the typeshed annotation in general, or special-case `dict.get` on `Top[dict[...]]`.
